File: SRM05A/SRM05A.c
#include <stdio.h>
int main() {
	int t;
	int numSystems, availableRepeaters, availableWires;
	int distance;
	scanf("%d",&t);
	while(t--) {
	    scanf("%d %d %d",&numSystems,&availableRepeaters,&availableWires);
	    for(int j = 0 ; j < numSystems-1 ; j++ ) {
	        scanf("%d",&distance);
	        if(distance > 150) {
	            availableRepeaters--;
	        }
	        availableWires -= distance;
	    }
	    if(availableWires < 0 || availableRepeaters < 0) {
	        printf("insufficient\n");
	    }
	    else {
	        printf("sufficient\n");
	    }
	}
	return 0;
}

File: SRM05A/SRM05A.py
t = int(input())
for i in range(0,t):
    temp = input().split(" ")
    numSystems = int(temp[0])
    availableRepeaters = int(temp[1])
    availableWires = int(temp[2])
    dist = input().split(" ")
    for j in range(0,numSystems-1):
        distance = int(dist[j])
        if(distance>150):
            availableRepeaters -= 1
        availableWires -= distance
    if(availableWires < 0 or availableRepeaters < 0):
        print("insufficient")
    else:
        print("sufficient")